pkg/crosscluster/physical/alter_replication_job.go

```diff
@@ -224,9 +228,33 @@ func alterReplicationJobHook(
 		}
 		jobRegistry := p.ExecCfg().JobRegistry
 		if alterTenantStmt.Producer {
+			if err := p.CheckPrivilege(
+				ctx, syntheticprivilege.GlobalPrivilegeObject, privilege.REPLICATIONSOURCE,
+			); err != nil {
+				return err
+			}
 			return alterTenantSetReplicationSource(ctx, p.InternalSQLTxn(), jobRegistry, options, tenInfo)
 		}
+		if err := p.CheckPrivilege(
+			ctx, syntheticprivilege.GlobalPrivilegeObject, privilege.REPLICATIONDEST,
+		); err != nil {
+			if tenInfo.PhysicalReplicationConsumerJobID == 0 {
+				return err
+			}
 
+			// If the user is attempting to alter an existing ingestion job that was
+			// created on an older version, they do not need the REPLICATIONDEST
+			// privilege, for now.
+			j, jobErr := jobRegistry.LoadJobWithTxn(ctx, tenInfo.PhysicalReplicationConsumerJobID, p.InternalSQLTxn())
+			if jobErr != nil {
+				return err
+			}
+			if j.Payload().CreationClusterVersion.Less(clusterversion.V25_2.Version()) {
+				p.BufferClientNotice(ctx, pgnotice.Newf("ALTER VIRTUAL CLUSTER REPLICATION cmds will require the REPLICATIONDEST privilege in a future release, even if the replication job was created on an older version"))
+				return nil
```

Comment:
Did you mean to return nil here? I would either expect this to return an error, or continue execution with a warning.

Reply:
ah nice catch! fixed.

Reply:
and added a test